output.c

```diff
@@ -523,8 +533,17 @@ void BulkUpload(Output_t* Out, u32 BufferIndex, u32 BufferCnt, u32 CPUID)
 	// not null es host
 	if (!s_IsESNULL)
 	{
-		int Sock = socket(AF_INET, SOCK_STREAM, 0);
-		assert(Sock > 0);
+		if (T->Sock > 0)
+		{
+			int error = 0;
+			socklen_t len = sizeof (error);
+			//printf("[%lu] Using fd: %d\n", pthread_self(), Sock);
+			if (getsockopt(T->Sock, SOL_SOCKET, SO_ERROR, &error, &len) == 0 && error == 0)
+				goto send_data;
```

Comment:
is there any other way besides a goto ? I really hate using goto in C code

Reply:
I will make if...else.. :)

Reply:
Removed "send_data" goto statement and replaced with new function and if condition.

Reply:
great thanks